### DISCUSSION TITLE
WARN cannot DNS resolve // ERROR doing http request - context deadline exceeded

### DISCUSSION THREAD

**Discussion by lonezel** [qdm12/ddns-updater#499] [Q&A] [answered]
**TLDR**: *Describe your issue in a one liner here*
I get a warning and errors when running the ddns-updater in a docker container and IP addresses are not updated. 

1. Is this urgent: Yes
2. DNS provider(s) you use: infomaniak
3. Program version:

Running version latest built on 2023-06-19T06:40:08.886Z (commit 6f75aa1)

4. What are you using to run the container: docker-compose

Logs:

```
2023-06-26T23:28:13+02:00 WARN cannot DNS resolve audiobooks.example.ch after 5 tries: lookup audiobooks.example.ch on (localhost-IP):53: read udp (docker-IP):43058->(localhost-IP):53: i/o timeout
2023-06-26T23:28:13+02:00 INFO IPv4 address of audiobooks.example.ch is invalid IP and your IPv4 address is (shows correct public IP)
```
also shows this error after a a few runs

```
2023-06-26T23:48:19+02:00 INFO Updating record [domain: example.ch | host: audiobooks | provider: infomaniak | ip: ipv4] to use (shows correct public IP)
2023-06-26T23:48:29+02:00 ERROR doing http request: Get "https://username:***@infomaniak.com/nic/update?hostname=audiobooks.example.ch": context deadline exceeded
```

Configuration file:

```{
  "settings": [
    {
      "provider": "infomaniak",
      "domain": "example.ch",
      "host": "audiobooks,nas,office,server",
      "username": "XXXX",
      "password": "XXXXX",
      "ip_version": "ipv4",
      "provider_ip": true
    }
  ]
}

```

Host OS:
Raspberry Pi OS Debian version: 11 (bullseye)

6. Additional Information: i am running a pi-hole as DNS server on the same raspberry pi

This is my first time setting it up, help would be appreciated.

**Comment by lonezel** [comment]
no clues anyone? :)

**Comment by qdm12** [comment, marked as answer]
`i/o timeout` means the DNS request timed out, and `context deadline exceeded` means the DNS went through but then the HTTP request/response timed out too (I think after 10s). Not much I can do, it's likely a networking reliability issue on your host/config. You could try using another DNS server using `RESOLVER_ADDRESS="1.1.1.1:53"` maybe that'll help.
